tests/api/v1/subjects/patchWithParent.js

```diff
@@ -0,0 +1,313 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/subjects/patchWithParent.js
+ */
+'use strict';
+
+const supertest = require('supertest');
+const api = supertest(require('../../../../index').app);
+const constants = require('../../../../api/v1/constants');
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const expect = require('chai').expect;
+const Subject = tu.db.Subject;
+const path = '/v1/subjects';
+const ZERO = 0;
+const ONE = 1;
+const TWO = 2;
+
+describe(`api: PATCH ${path} with parents`, () => {
+  let token;
+  const n0 = { name: `${tu.namePrefix}Canada`, isPublished: true };
+  const n1 = { name: `${tu.namePrefix}Ontario`, isPublished: true };
+  const n2 = { name: `${tu.namePrefix}Manitoba`, isPublished: true };
+  const p0 = { name: `${tu.namePrefix}NA`, isPublished: true };
+  const p1 = { name:
```

Comment:
A tip: Its better to use logical names  instead of 'n0' 'n1's etc, so that they are more readable and a new user can understand the code in just a look. Or add some comments/examples explaining the hierarchy in your mind